mailet/standard/src/main/java/org/apache/james/transport/mailets/ToProcessor.java

```diff
@@ -42,62 +44,46 @@
  */
 public class ToProcessor extends GenericMailet {
 
-    /**
-     * Controls certain log messages
-     */
-    private boolean isDebug = false;
+    private boolean debug;
+    private String processor;
+    private Optional<String> noticeText;
 
-    /**
-     * The name of the processor to which this mailet forwards mail
-     */
-    String processor;
-
-    /**
-     * The error message to attach to the forwarded message
-     */
-    String noticeText = null;
-
-    /**
-     * Initialize the mailet
-     *
-     * @throws MailetException if the processor parameter is missing
-     */
+    @Override
     public void init() throws MailetException {
-        isDebug = (getInitParameter("debug") == null) ? false : Boolean.valueOf(getInitParameter("debug"));
+        debug = isDebug();
         processor = getInitParameter("processor");
         if (processor == null) {
             throw new MailetException("processor parameter is required");
         }
-        noticeText = getInitParameter("notice");
+        noticeText = Optional.fromNullable(getInitParameter("notice"));
     }
 
-    /**
-     * Deliver a mail to the processor.
-     *
-     * @param mail the mail to process
-     *
-     * @throws MessagingException in all cases
-     */
+    private boolean isDebug() {
+        String parameter = getInitParameter("debug");
+        return (parameter == null) ? false : Boolean.valueOf(parameter);
+    }
+
+    @Override
+    public String getMailetInfo() {
+        return "ToProcessor Mailet";
+    }
+
+    @Override
     public void service(Mail mail) throws MessagingException {
-        if (isDebug) {
+        if (debug) {
             log(String.format("Sending mail %s to %s", mail, processor));
         }
         mail.setState(processor);
-        if (noticeText != null) {
+        setErrorMessage(mail);
```

Comment:
Naming here is confusing... The way I understand it is it set an error message on every mail. I need to read the actual method to understand that it is only setted if the notice is present.

Maybe wrapping call in if block will make it easier to read.
Maybe naming is enough (setOptionalErrorMessage)

By the way, is it a notice or an error message ?